ladybug/windrose.py

```diff
@@ -195,56 +195,68 @@ def frequency_spacing_distance(self):
         return self._frequency_spacing_distance
 
     @frequency_spacing_distance.setter
-    def frequency_spacing_distance(self, frequency_spacing_distance):
-        f = frequency_spacing_distance
-        assert f > 0.0 and isinstance(f, (float, int)), 'frequency_spacing_distance' \
-            'should be a number greater then 0. Got {}.'.format(f)
+    def frequency_spacing_distance(self, val):
+        assert val > 0.0 and isinstance(val, (float, int)), \
+            'frequency_spacing_distance should be a number greater then 0. ' \
+            'Got {}.'.format(val)
         self._compass = None
         self._container = None
-        self._frequency_spacing_distance = frequency_spacing_distance
+        self._frequency_spacing_distance = val
 
     @property
+    def frequency_spacing_hypot_distance(self):
+        """Length of radius of windrose assuming perpendicular frequency spacing.
+
+        The limit of the frequency_spacing_hypot_distance as number of directions
+        approaches infinity equals the frequency_spacing_distance (as is the case
+        with a circle).
+        """
+        _theta = 360.0 / self._number_of_directions
+        theta = _theta / 2.0 / 180.0 * math.pi
+        return self.frequency_spacing_distance / math.cos(theta)
+
+    @ property
     def frequency_maximum(self):
         """Get maximum frequency to represent in the windrose plot.
 
         Default: number of items in largest windrose histogram bin.
         """
         return self.frequency_intervals_compass * self.frequency_hours
 
-    @property
+    @ property
```

Comment:
All of these extra whitespaces are unnecessary and I think they are not PEP8.

Reply:
Yeah this seems to be some overly aggressive linting from VSCode. Might be autopep8 again. I can try and disable this.

Reply:
Perhaps you are using a different styling standard that's different than PEP8. But we definitely don't need the space to be PEP8 compliant.